ghproxy/ghcache/ghcache.go

```diff
@@ -202,39 +211,55 @@ func (u upstreamTransport) RoundTrip(req *http.Request) (*http.Response, error)
 // NewDiskCache creates a GitHub cache RoundTripper that is backed by a disk
 // cache.
 func NewDiskCache(delegate http.RoundTripper, cacheDir string, cacheSizeGB, maxConcurrency int) http.RoundTripper {
-	return NewFromCache(delegate, diskcache.NewWithDiskv(
-		diskv.New(diskv.Options{
-			BasePath:     path.Join(cacheDir, "data"),
-			TempDir:      path.Join(cacheDir, "temp"),
-			CacheSizeMax: uint64(cacheSizeGB) * uint64(1000000000), // convert G to B
-		})),
+	return NewFromCache(delegate,
+		func(partitionKey string) httpcache.Cache {
+			return diskcache.NewWithDiskv(
+				diskv.New(diskv.Options{
+					BasePath:     path.Join(cacheDir, "data", partitionKey),
+					TempDir:      path.Join(cacheDir, "temp", partitionKey),
+					CacheSizeMax: uint64(cacheSizeGB) * uint64(1000000000), // convert G to B
+				}))
+		},
 		maxConcurrency,
 	)
 }
 
 // NewMemCache creates a GitHub cache RoundTripper that is backed by a memory
 // cache.
```

Comment:
This does or does not support partitioning?

Reply:
It does, it just doesn't need to have a parition key for that

Reply:
Could you leave a comment to that effect? It's not obvious from a read of the code

Reply:
done